sdk/storage/src/blobs/blob_client.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "blobs/blob_client.hpp"
+
+#include "common/storage_common.hpp"
+#include "http/curl/curl.hpp"
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  BlobClient BlobClient::FromConnectionString(
+      const std::string& connectionString,
+      const std::string& containerName,
+      const std::string& blobName,
+      const BlobClientOptions& options)
+  {
+    auto parsedConnectionString = ParseConnectionString(connectionString);
+
+    std::string accountName;
+    std::string accountKey;
+    std::string blobEndpoint;
+    std::string EndpointSuffix;
+    std::string defaultEndpointsProtocol = ".core.windows.net";
+
+    auto ite = parsedConnectionString.find("AccountName");
+    if (ite != parsedConnectionString.end())
+    {
+      accountName = ite->second;
+    }
+    ite = parsedConnectionString.find("AccountKey");
+    if (ite != parsedConnectionString.end())
+    {
+      accountKey = ite->second;
+    }
+    ite = parsedConnectionString.find("BlobEndpoint");
+    if (ite != parsedConnectionString.end())
+    {
+      blobEndpoint = ite->second;
+    }
+    ite = parsedConnectionString.find("EndpointSuffix");
+    if (ite != parsedConnectionString.end())
+    {
+      EndpointSuffix = ite->second;
+    }
+    ite = parsedConnectionString.find("DefaultEndpointsProtocol");
+    if (ite != parsedConnectionString.end())
+    {
+      defaultEndpointsProtocol = ite->second;
+    }
+
+    UrlBuilder builder;
+    builder.SetScheme(defaultEndpointsProtocol);
+    if (!blobEndpoint.empty())
+    {
+      builder = UrlBuilder(blobEndpoint);
```

Comment:
>builder = UrlBuilder(blobEndpoint); [](start = 6, length = 35)

Consider the below connection string:
```
# All service endpoints
DefaultEndpointsProtocol=https;
BlobEndpoint=[http://www.mydomain.com](http://www.mydomain.com);
FileEndpoint=[https://myaccount.file.core.windows.net](https://myaccount.file.core.windows.net);
QueueEndpoint=[https://myaccount.queue.core.windows.net](https://myaccount.queue.core.windows.net);
TableEndpoint=[https://myaccount.table.core.windows.net](https://myaccount.table.core.windows.net);
AccountName=storagesample;
AccountKey=<account-key>
```
The UrlBuilder should do something different with blob's endpoint, as the account url should be:
```
[http://www.mydomain.com/storagesample/](http://www.mydomain.com/storagesample/)
```
See [https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string)

Reply:
Mark this as TODO if customized blob endpoint is not supported for now.

---
In reply to: [433676313](https://github.com/Azure/azure-sdk-for-cpp/pull/144#discussion_r433676313) [](ancestors = 433676313)

Reply:
no, it's not. But there are still many kinds of connection strings we are going to support in the future.